File: python/EPI/reversePolishNotation.py
def RPN(s):
    result  = []
    operations = {
        '+': lambda x,y: x + y,
        '*': lambda x,y: x * y,
        '/': lambda x,y: y / x,
        '-': lambda x,y: y - x,
    }


    for token in s.split(' '):
        if(token in operations):
            result.append(operations[token](result.pop(), result.pop()))
        else:
            result.append(int(token))
    return result

print(RPN('3 4 +'))

File: javascript/EPI/reversePolishNotation.js



const RPN = (s) =>{
    const arr = s.split(' ');
    const operations = {
        '+': (x,y)=> x + y,
        '*': (x,y)=> x * y,
        '/': (x,y)=> y - x,
        '-': (x,y)=> y - x,
    }
    const result = [];
    for(const token of arr){
        !isNaN(parseInt(token))
        ? result.push(parseInt(token))
        : result.push(operations[token](result.pop(), result.pop()))
    }
    return result;
}

console.log(RPN('3 4 + 2 * 1 +'))